awscli/clidriver.py

```diff
@@ -444,8 +424,8 @@ def main(self, args=None):
                 'CLI_VERSION', self.session.user_agent(), 'CLI')
             HISTORY_RECORDER.record('CLI_ARGUMENTS', args, 'CLI')
             return command_table[parsed_args.command](remaining, parsed_args)
-        except Exception as e:
-            return self._error_handlers_chain.handle_exception(
+        except BaseException as e:
```

Comment:
Is there a particular reason/edge case we are catching `BaseException`? In general, `Exception` should be the base class for all potential errors that are thrown and is typically what we always catch if we want to catch all exceptions.

Reply:
KeyboardInterrupt inherited from BaseException

Reply:
Gotcha. Let's keep it at `BaseException` then